### PR TITLE
feat: tree folder visualization

### PR DESCRIPTION
**Pull request by RiccardoCafa**
This is a feature that shows the hierarchy for the current folder, I've named it "missogates" and used an airplane cause is like going somewhere and using a specific gate to reach the destination.

To use is very simple, just click on the airplane then should appear a hyperlink for every children's folder for the current one. If none then a message should appear telling you there are no gates. Click it again to make it disappear.

### CODE REVIEW

File: src/components/Pad.tsx [hunk 1]
```diff
@@ -100,12 +104,21 @@ function Pad() {
...
+        >
+          ✈️{missogates ? " missogates" : ""}
+        </h3>
         <h2 className="logo" onClick={() => setOnlyView(!onlyView)}>
           MISSOPAD{" "}
         </h2>
         {onlyView && "👀"}
       </header>
 
+      {missogates ? <Tree /> : <></>}
```
Comment (oenzoferrari) [inline src/components/Pad.tsx:120]:
```suggestion
      {missogates && <Tree />}
```

File: src/components/Pad.tsx [hunk 2]
```diff
@@ -31,6 +33,8 @@ function Pad() {
 
   const [onlyView, setOnlyView] = useState<boolean>(false);
 
+  const [missogates, setMissogates] = useState<boolean>(false);
```
Comment (oenzoferrari) [inline src/components/Pad.tsx:36]:
```suggestion
  const [showMissogates, setShowMissogates] = useState<boolean>(false);
```

File: src/components/Pad.tsx [hunk 3]
```diff
@@ -100,12 +104,21 @@ function Pad() {
           padding: "2vh",
         }}
       >
+        <h3
+          style={{ position: "absolute", top: -10, left: 10, zIndex: 9999 }}
+          className="logo"
+          onClick={() => setMissogates(!missogates)}
+        >
+          ✈️{missogates ? " missogates" : ""}
```
Comment (oenzoferrari) [inline src/components/Pad.tsx:112]:
```suggestion
          onClick={() => setShowMissogates(!showmMissogates)}
        >
          ✈️{showMissogates && " missogates"}
```

File: src/components/Pad.tsx (hunk 1 above)
Comment (oenzoferrari) [reply src/components/Pad.tsx:120]: P.s.: not sure if this works

File: src/components/Tree.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  function getDataSnapshotArray(node: DataSnapshot): DataSnapshot[] {
+    const childs: DataSnapshot[] = [];
+    node.forEach((child) => {
+      if (child.size > 0) {
+        childs.push(child);
+      }
+    });
+    return childs;
+  }
```
Comment (oenzoferrari) [inline src/components/Tree.tsx:28]:
```suggestion
    const children: DataSnapshot[] = [];
    node.forEach((child) => {
      if (child.size > 0) {
        childs.push(child);
      }
    });
    return children;
  }
```

File: src/components/Tree.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+              display: "flex",
+              flexDirection: "column",
+              whiteSpace: "nowrap",
+            }}
+          >
+            {mountTree(route)}
+          </div>
+        );
+      }
+    });
```
Comment (oenzoferrari) [inline src/components/Tree.tsx:45]:
```suggestion
    return routes.filter(route => route.size > 0).map((route) => {
        return (
          <div
            style={{
              display: "flex",
              flexDirection: "column",
              whiteSpace: "nowrap",
            }}
          >
            {mountTree(route)}
          </div>
        );
    });
```

File: src/components/Tree.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        href={`${pathname}${subpath}\\${route.key}`}
+        style={{
+          color: "white",
+          marginBottom: "0.2em",
+          overflowWrap: "break-word",
+          textDecoration: "none",
+          marginLeft: level - 1 + "em",
+        }}
+      >
+        {(level === 1 ? ".\\" : "↳ ") + route.key}
```
Comment (oenzoferrari) [inline src/components/Tree.tsx:64]: I think these should be constants, something like:

```js
ROOT_SYMBOL = ".\\"
CHILD_SYMBOL = "↳ "
```

File: src/components/Tree.tsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+    return (
+      <>
+        {getRouteLink(node, subpath, recursion++)}
+        {childs.map((route) => {
+          return mountTree(route, subpath + "/" + node.key, recursion);
+        })}
+      </>
+    );
+  }
```
Comment (oenzoferrari) [inline src/components/Tree.tsx:86]:
```suggestion
    const children: DataSnapshot[] = getDataSnapshotArray(node);

    return (
      <>
        {getRouteLink(node, subpath, recursion++)}
        {children.map((route) => {
          return mountTree(route, subpath + "/" + node.key, recursion);
        })}
      </>
    );
  }
```

File: src/components/Tree.tsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        height: "100%",
+        top: "0",
+        width: "30%",
+        overflowY: "scroll",
+        overflowX: "hidden",
+        backgroundColor: "#0d0f17",
+      }}
+    >
+      {getTree()}
+      {routes.length == 0 ? <p>no gates here :(</p> : <></>}
```
Comment (oenzoferrari) [inline src/components/Tree.tsx:104]:
```suggestion
      {routes.length == 0 && <p>no gates here :(</p>}
```

P.s.: Not sure if this works

File: src/components/Pad.tsx (hunk 3 above)
Comment (oenzoferrari) [reply src/components/Pad.tsx:112]: P.s.: not sure if this works

**Review by oenzoferrari — CHANGES_REQUESTED** [review]
Very cool, I suggested some minor style improvements 👍

File: src/components/Tree.tsx (hunk 5 above)
Comment (RiccardoCafa) [reply src/components/Tree.tsx:45]: It doesn't work without returning JSX

**Review by oenzoferrari — APPROVED** [review]

**Pull request merged**